develop-seat-adjuster.md

```diff
@@ -0,0 +1,147 @@
+and the configuration of a [development environment in a container](https://github.com/SoftwareDefinedVehicle/seat-adjuster-template/tree/main/.devcontainer).
+
+In addition to the Python template used in this guide, there is a general [C++ template](https://github.com/eclipse-velocitas/vehicle-app-cpp-template) available.
+
+## Execute Development Container in VSCode
+
+You then checkout the repository on your development machine and open it in VSCode. From VSCode
+you can execute the development container (DevContainer) configured in the repository. For this to function, you need the following tools to be installed on your computer:
+
+- Docker (e.g. [Docker Desktop](https://www.docker.com) or [Rancher Desktop](https://rancherdesktop.io))
+- [VSCode](https://code.visualstudio.com) with the
+[Dev Containers](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-containers) plugin installed.
+
+In the best case, VSCode may detect that the repository contains a DevContainer configuration and ask you whether to execute it.
+Alternatively, you can press `F1` in VSCode and then enter `DevContainers: Reopen in container` to start the DevContainer.
+In either case, VSCode should reopen and then connect to the DevContainer.
+Because of the DevContainer, we do not need to make further modifications to the developer machine.
+
+## Develop the application
+
+In the next step, we develop the actual application. You find the skeleton code in `/app/src/main.py`.
+
+Let us walk through some lines of the code where the logic is in the `vapp.py` while the entry point is in `main.py`.
+
+In the `onStart(self)` function you should subscribe to any changes of the current value of the VSS signal `Vehicle.Cabin.Seat.Row1.Pos1.Position` in the KUKSA Databroker and provide the `on_seat_position_changed` method as callback.
+
+```python
+async def on_start(self):
+    """Run when the vehicle app starts"""
+    # TODO subscribe to Vehicle.Cabin.Seat.Row1.Pos1.Position and provide
+    # on_seat_position_changed as callback.
+```
+
+When the seat position changes, the `on_seat_position_changed` function is triggered and should publish the new seat position value as MQTT-message to the topic `seatadjuster/currentPosition`.
+
+```python
+async def on_seat_position_changed(self, data: DataPointReply):
+    # TODO publish the current position as MQTT message to CURRENT_POSITION_TOPIC.
+```
+
+In addition, we have the function `on_set_position_request_received` which is triggered for every MQTT-message to the topic `seatadjuster/setPosition/request`.
+If the vehicle speed is zero, meaning that the vehicle does not move, you should set the target value in the KUKSA Databroker for the position signal of the seat
+to the value requested in the incoming MQTT message.
+
+```python
+ @subscribe_topic("seatadjuster/setPosition/request")
```

Comment:
Just as an idea: Leave the topics empty in the skeleton code so that the user has to fill them in?

Reply:
hmm don't know if that is really important, I mean does it help to set different topics? These are just arbitrary strings without a big meaning or any logic behind. So defining them just costs time and I don't see any benefit for the developer.

Reply:
it's ok how it is, was just an idea